napari_plugin_manager/qt_plugin_dialog.py

```diff
@@ -678,15 +686,23 @@ def tag_unavailable(self, metadata: npe2.PackageMetadata):
             widget.action_button.setEnabled(False)
             widget.warning_tooltip.setVisible(True)
 
-    def filter(self, text: str):
+    def filter(self, text: str, starts_with_chars: int = 1):
         """Filter items to those containing `text`."""
         if text:
             # PySide has some issues, so we compare using id
             # See: https://bugreports.qt.io/browse/PYSIDE-74
-            shown = [
-                id(it)
-                for it in self.findItems(text, Qt.MatchFlag.MatchContains)
-            ]
+            flag = (
+                Qt.MatchFlag.MatchStartsWith
+                if len(text) <= starts_with_chars
+                else Qt.MatchFlag.MatchContains
+            )
+            shown = [id(it) for it in self.findItems(text, flag)]
+            if len(text) <= starts_with_chars:
+                text2 = f'napari-{text}'
+                shown = set(
+                    [id(it) for it in self.findItems(text2, flag)] + shown
+                )
```

Comment:
```suggestion
            if len(text) <= starts_with_chars:
                flag = Qt.MatchFlag.MatchStartsWith
                queries = (text, f'napari-{text}')
            else:
                flag = Qt.MatchFlag.MatchContains
                queries = (text,)
            shown = {id(it) for query in queries for it in self.findItems(query, flag)}
```

What about this refactor? :)

Reply:
Not a fan of double nested comprehensions (it becomes a bit unreadable) but sure why not :)